packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -712,11 +711,27 @@ export class ScrollableTabBar extends TabBar<Widget> {
     protected openTabsContainer: HTMLDivElement;
     protected openTabsRoot: Root;
 
-    constructor(options?: TabBar.IOptions<Widget> & PerfectScrollbar.Options, dynamicTabOptions?: ScrollableTabBar.Options) {
+    constructor(options?: TabBar.IOptions<Widget>, protected readonly scrollbarOptions?: PerfectScrollbar.Options, dynamicTabOptions?: ScrollableTabBar.Options) {
```

Comment:
I agree with this change, but it is technically breaking. Is it worth the breakage?

Reply:
Yes, I think it is: the code has a better separation of concerns and since clients will have to change their code, it's preferable that the get a compile time error over keeping the code type compatible but breaking at runtime. How to fix the problem is very straightforward.

Reply:
In that case, don't forget to add it to the changelog :-).